Tanga/Features/Notifications/AppDelegateNotificationsExtension.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  AppDelegateNotificationsExtension.swift
+//  Tanga
+//
+//  Created by Rygel Louv on 29/01/2025.
+//
+
+import SwiftUI
+import FirebaseMessaging
+import UserNotifications
+import OSLog
+
+
+extension AppDelegate: MessagingDelegate {
+    func messaging(_ messaging: Messaging, didReceiveRegistrationToken fcmToken: String?) {
+
+      let deviceToken:[String: String] = ["token": fcmToken ?? ""]
+        print("Device token: ", deviceToken)
+        
+        Messaging.messaging().subscribe(toTopic: "ios_test_topic") { error in
+            if let error = error {
+                print("Error subscribing to topic: \(error)")
+            } else {
+                Logger.notifications.info("Successfully subscribed to topic: ios_test_topic")
+            }
+        }
+    }
+}
+
+@available(iOS 10, *)
+extension AppDelegate : UNUserNotificationCenterDelegate {
+
+  // Receive displayed notifications for iOS 10 devices.
+   func userNotificationCenter(_ center: UNUserNotificationCenter,
+                              willPresent notification: UNNotification,
+        withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
+        let userInfo = notification.request.content.userInfo
+
+        if let messageID = userInfo[gcmMessageIDKey] {
+            //TODO: we should only log this in debug/dev and never in production
+            print("willPresent -> Message ID: \(messageID)")
+        }
+
+        completionHandler([[.banner, .badge, .sound]])
+   }
+
+    func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+        Logger.notifications.info("Registered for apple notification")
+        Messaging.messaging().setAPNSToken(deviceToken, type: .unknown)
+    }
+
+    func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
+        Logger.notifications.info("Falied to register for apple notification")
+    }
+
+    func userNotificationCenter(_ center: UNUserNotificationCenter,
+                              didReceive response: UNNotificationResponse,
+                              withCompletionHandler completionHandler: @escaping () -> Void) {
+      let userInfo = response.notification.request.content.userInfo
+
+      if let messageID = userInfo[gcmMessageIDKey] {
+          //TODO: we should only log this in debug/dev and never in production
+          print("Message ID from userNotificationCenter didReceive: \(messageID)")
```

Comment:
Note: Unable to use Logger with string interpolation. Having compile time issues - no idea how to fix for now